src/client/game.cpp

```diff
@@ -2689,14 +2691,22 @@ void Game::updatePlayerControl(const CameraOrientation &cam)
 
 	//TimeTaker tt("update player control", NULL, PRECISION_NANO);
 
+
+	if (g_settings->getBool("toggle_sneak")) {
+		if (wasKeyPressed(KeyType::SNEAK)) runData.isSneaking = !runData.isSneaking;
+	} else {
+		runData.isSneaking = isKeyDown(KeyType::SNEAK);
+	}
+
+
 	PlayerControl control(
 		isKeyDown(KeyType::FORWARD),
 		isKeyDown(KeyType::BACKWARD),
 		isKeyDown(KeyType::LEFT),
 		isKeyDown(KeyType::RIGHT),
 		isKeyDown(KeyType::JUMP) || player->getAutojump(),
 		isKeyDown(KeyType::AUX1),
-		isKeyDown(KeyType::SNEAK),
+		runData.isSneaking,
```

Comment:
Not sure about this. Minetest is being inconsistent here. Controls are both "kind of" keybinds and "abstract" controls. The "forward" control for example does not take autoforward into account, while the "jump" control does take autojump into account.

I'm not sure whether we should take "sneak toggling" into account here. On the one hand, this is unsuitable for the use case of overloading actions using sneak (for example sneak + rightclick placing a node rather than opening a chest). On the other hand, how else can servers detect whether a player is presently sneaking?

Perhaps an option would be to separate this out into two fields (though the question of which of these to call `sneak` would persist)?

Reply:
@appgurueu any updates? Unsubscribing

Reply:
I think that it's good as is - separating them makes the code more complicated and more confusing. You want to know whether the player is commanding a sneak, doesn't matter whether it's done by a key down, toggle, mouse button, or whatever

Reply:
Can be left as-is.